vtex/address-form
# CHK-2532: Fix NPM imports of external ESM modules without interop

#### What is the purpose of this pull request?

This PR adds the `interop: "auto"` to the `output` config of our Rollup build.

[Rollup documentation](https://rollupjs.org/configuration-options/#output-interop)

The option with value of `"auto"` mimics TypeScript behavior when using the `allowSyntheticDefaultImports` compiler option, which is the expected behavior for this repository.

#### What problem is this solving?

It fixes the imports of external ESM packages, such as `@vtex/styleguide` which weren't working on the previously published NPM versions.

This will unblock repositories that need to use the latest version of `address-form`, but need to use the NPM package instead of VTEX IO.

The diff of the output should be similar to the following:

```diff
 var React = require("react");
 var Spinner = require("@vtex/styleguide/lib/Spinner");

+function _interopDefault(e) { return e && e.__esModule ? e : { default: e }; }
+
+var React__default = _interopDefault(React);
+var Spinner__default = _interopDefault(Spinner);
+
 const SpinnerLoader = () => {
-   return React.createElement(Spinner, { size: 15 });
+   return React__default.default.createElement(Spinner__default.default, { size: 15 });
 }
```

#### How should this be manually tested?

Run `yarn build` and open the file `react/lib/Spinner.js`. The file should use the `interop` function on the ESM imports.

#### Screenshots or example usage

N/A

#### Types of changes
